### DISCUSSION TITLE
Global site tag scripts inject GA script above head tag

### DISCUSSION THREAD

**Discussion by aldrichdev** [vercel/next.js#39082] [Help] [answered]
Hi, I am having a very bizarre issue with a NextJS site. I have the following code in a component that adds scripts for GTag, Google Analytics and DoubleClick/Floodlight. This component gets rendered in the main `_app.tsx`:

```
    <Script async src="https://www.googletagmanager.com/gtag/js?id=UA-0000000-1" />
    <Script async src="https://www.googletagmanager.com/gtag/js?id=DC-0000000" />
    <Script
      id="gtagConfig"
      strategy="afterInteractive"
      dangerouslySetInnerHTML={{
        __html: `window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());
          gtag('config', 'UA-0000000-1');
          gtag('config', 'DC-0000000');
          gtag('config', 'AW-00000000');`,
      }}
    />
```

The 3rd script seems to create and inject a `https://www.google-analytics.com/analytics.js` script, among others. The issue is, it is injecting this script **above** the `<head>` (in the `<html>` element), and that script needs to be inside the `<head>` per our analytics team, or Google can't read it.

![image](https://user-images.githubusercontent.com/17816529/181348184-4664f011-daae-4076-9c0b-bf1e8552a8a8.png)

I've tried the following alternatives but they do not work:

- Adding `strategy="beforeInteractive"` to all 3 scripts. The 1st two get added to the head, but the 3rd does not get added at all (`dangerouslySetInnerHTML` and `beforeInteractive` don't mix, from what I've read).
- Using regular `<script>` tags instead, like below. This puts those in the `<head>` but it still injects the `analytics.js` script above the head.
    ```<Head>
      <script async src="https://www.googletagmanager.com/gtag/js?id=UA-0000000-1" />
      <script async src="https://www.googletagmanager.com/gtag/js?id=DC-0000000" />
      <script dangerouslySetInnerHTML={{
        __html: `window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());
          gtag('config', 'UA-0000000-1');
          gtag('config', 'DC-0000000');
          gtag('config', 'AW-00000000');`,
      }} />
    </Head>```

Can anyone clarify if this is a NextJS implementation issue or if this is a bug on the `gtag` API side?

**Comment by aldrichdev** [comment, marked as answer]
Actually, this looks like it will not be a problem once we switch over to GA4 (from Universal Analytics). I updated the code and the `analytics.js` file does not get added anymore (also, no Gtag/Google related tags are getting added above the head). We were planning on switching to GA4 anyway so it seems like that solves the issue.

**Reply by PTBYSR** [reply]
Is it possible to edit code straight from the vercel dashboard without needing to go to the code base? I want to add a scripts for GTM but i don't have access to the code base just the vercel dashboard
